File: spoj/toandfro.py
#Accepted!
#https://www.spoj.pl/problems/TOANDFRO/
#0.03 	3.7M

import sys

def reverse(s):
	#equivalente: s = s[::-1]
	s2 = ""
	for i in range(len(s)-1, -1, -1):
		s2 += s[i]
	return s2

n = input()
while n != 0:
	s = raw_input()
	i = 0;
	l = []
	bien = True
	while i < len(s):
		aux = s[i:i+n]
		if not bien:
			aux = reverse(aux)
		l += [aux]
		bien = not bien
		i += n
	for c in range(n):
		for f in range(len(l)):
			sys.stdout.write(l[f][c])
	print
	n = input()

File: spoj/toandfro.cpp
//Accepted!
//https://www.spoj.pl/problems/TOANDFRO/
//2012-01-13 20:34:35
//0.01 	2.8M

#include <iostream>
#include <algorithm>
#include <sstream>
#include <vector>

using namespace std;

int main(){
	int n;
	string s, aux;
	while (cin>>n && n != 0){
		cin.ignore();
		getline(cin, s);
		vector<string> v;
		bool bien = true;
		int i=0;
		while (i < s.length()){
			aux = s.substr(i, n);
			if (!bien)
				reverse(aux.begin(), aux.end());
			v.push_back(aux);
			bien = !bien;
			i += n;
		}
		
		stringstream ss;
		for (int c=0; c<n; ++c){
			for (int f=0; f<v.size(); ++f)
				ss<<v[f][c];
		}
		cout<<ss.str()<<endl;
	}
	return 0;
}
